crates/environ/src/component/translate.rs

```diff
@@ -898,7 +890,7 @@ impl<'a, 'data> Translator<'a, 'data> {
                     }
 
                     // ignored during this type pass
-                    ComponentItem::Module(_) => {}
+                    ComponentItem::Module(_) | ComponentItem::Type => {}
```

Comment:
I think that this may not be quite right to ignore because this should add an item to the type namespace but this isn't doing that? I think that means that types defined after this `alias` wouldn't be numbered correctly.

Reply:
That's a good point. Here's what I think might solve this:

1. When we pop a type scope (i.e. finish parsing a component), associate it with the current translation to signify its statically-known types.
2. Add a `ComponentTypeIndex` to the `ComponentItem::Type` variant.
3. When processing an alias to an exported type, lookup the `TypeDef` via the referenced translation's types and push it to the current type scope.

Does that sound about right?

Reply:
Thinking about this and also from our discussion in video just now, I think what I'm getting at is that I would naively expect this to be `enum ComponentItem { Type(TypeDef) }` where unlike the other variants of `ComponentItem` types are "fully resolved" to a final type information which means we don't need to keep around the type scope information. Otherwise my worry is that when you import and instance with a type export and then alias a type from that during the `translate.rs` phase we don't otherwise have a type scope to associate with that because we don't know how the import was satisfied.

Reply:
Ah yeah, that makes more sense, no need to look up by index here, just push the `TypeDef` into the local type scope. I'll push up the fix.

Reply:
Pushed up the fix. Working on test cases to these two PRs now.